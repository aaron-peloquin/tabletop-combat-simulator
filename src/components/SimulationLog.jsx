import React, {Fragment} from "react"
import {connect} from "react-redux"
import {
  Table,
  TableHead, TableBody,
  TableRow, TableCell,
  Typography,
} from "@material-ui/core"
import {withStyles} from "@material-ui/core/styles"

const Styles = () => {
  return {
    LogItems: {
      textAlign: "left",
    },
  }
}

const SimulationLog = (props) => {
  const {classes, Log} = props
  return (Log.length > 0?<Fragment>
    <Typography variant="h4">Combat Log</Typography>
    <Table className={classes.LogItems}>
      <TableHead>
        <TableRow>
          <TableCell>Round: Creature</TableCell>
          <TableCell>Message</TableCell>
        </TableRow>
      </TableHead>
      <TableBody>
        {Log.map((Item, k) => {
          return <TableRow key={k}>
            <TableCell>{Item.Round}: {Item.Creature} ({Item.Team.toUpperCase()})</TableCell>
            <TableCell>{Item.Message}</TableCell>
          </TableRow>
        })}
      </TableBody>
    </Table>
  </Fragment>:<Fragment />)
}

const MapStateToProps = (state) => {
  let Log = []
  if (state.combat.Victory.length > 0) {
    Log = state.combat.Log
  }

  return {
    Log,
  }
}

const ConnectedComponent = connect(MapStateToProps)(SimulationLog)
const StyledComponent = withStyles(Styles)(ConnectedComponent)

export default StyledComponent
