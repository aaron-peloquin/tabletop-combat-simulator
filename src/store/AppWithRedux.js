import React from 'react'
import {initializeStore} from './store'

const isServer = typeof window === 'undefined'
const reduxStoreSymbol = Symbol("reduxStore")

function getOrCreateStore (initialState) {
  // Always make a new store if server, otherwise state is shared between requests
  if (isServer) {
    return initializeStore(initialState)
  }

  // Create store if unavailable on the client and set it on the window object
  if (!window[reduxStoreSymbol]) {
    window[reduxStoreSymbol] = initializeStore(initialState)
  }
  return window[reduxStoreSymbol]
}

export default (App) => {
  return class AppWithRedux extends React.Component {
    static async getInitialProps (appContext) {
      // Get or Create the store with `undefined` as initialState
      // This allows you to set a custom default initialState
      const reduxStore = getOrCreateStore()

      // Provide the store to getInitialProps of pages
      appContext.ctx.reduxStore = reduxStore

      let appProps = {}
      if (typeof App.getInitialProps === 'function') {
        appProps = await App.getInitialProps(appContext)
      }

      return {
        ...appProps,
        initialReduxState: reduxStore.getState()
      }
    }

    constructor (props) {
      super(props)
      this.reduxStore = getOrCreateStore(props.initialReduxState)
    }

    render () {
      return <App {...this.props} reduxStore={this.reduxStore} />
    }
  }
}
