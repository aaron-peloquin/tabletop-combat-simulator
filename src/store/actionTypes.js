/**
 * This is an object of Redux Actions. It mostly exists to
 *  ensure we do not create duplicate actions, but it can
 *  also be used as a list of our existing actions
*/
const actions = {
  CopyCreature: 'CopyCreature',
  DeleteAllCreatures: 'DeleteAllCreatures',
  DeleteCreature: 'DeleteCreature',
  SaveCreature: 'SaveCreature',
  SetEditCreature: 'SetEditCreature',
  ToggleSidebar: 'ToggleSidebar',
  RunSimulation: 'RunSimulation',
  UpdateEditCreature: 'UpdateEditCreature',
}

export default actions
