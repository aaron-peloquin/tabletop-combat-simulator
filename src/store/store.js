import {createStore, applyMiddleware, combineReducers} from "redux"
import {composeWithDevTools} from "redux-devtools-extension"
import thunkMiddleware from "redux-thunk"

/** Reducers */
import combat from "./reducer/combat"
import creatures from "./reducer/creatures"
import sideBar from "./reducer/sideBar"
import editing from "./reducer/editing"

/** Reducers */
const reducers = combineReducers({
  combat,
  creatures,
  editing,
  sideBar,
})

/**
 * Initializes the store
 * @param {obj} initialState the state our app starts with
 * @return {obj} the redux store
 */
export function initializeStore(initialState = initialState) {
  return createStore(reducers, initialState, composeWithDevTools(applyMiddleware(thunkMiddleware)))
}
