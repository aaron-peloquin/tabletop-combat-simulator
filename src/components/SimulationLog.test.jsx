import {Provider} from "react-redux"
import {createShallow} from "@material-ui/core/test-utils"
import toJson from "enzyme-to-json"
import {initializeStore} from "../store/store"

import SimulationLog from "./SimulationLog"

describe("<SimulationLog />", () => {
  const renderShallowUntilComponent = createShallow({"untilSelector": "SimulationLog"})
  let store; let props; let SimulationLogComponent

  beforeEach(() => {
    store = initializeStore({})

    SimulationLogComponent = renderShallowUntilComponent(<Provider store={store}>
      <SimulationLog />
    </Provider>)
  })

  it("renders", () => {
    expect(typeof SimulationLogComponent).toBe("object")
  })

  it("snapshots", () => {
    const Tree = toJson(SimulationLogComponent)
    expect(Tree).toMatchSnapshot()
  })
})
