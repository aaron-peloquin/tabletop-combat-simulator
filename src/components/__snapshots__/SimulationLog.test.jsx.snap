// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimulationLog /> snapshots 1`] = `
<Fragment>
  <WithStyles(Typography)
    variant="h4"
  >
    Combat Log
  </WithStyles(Typography)>
  <WithStyles(Table)
    className="Connect-SimulationLog--LogItems-1"
  >
    <WithStyles(TableHead)>
      <WithStyles(TableRow)>
        <WithStyles(TableCell)>
          Round: Creature
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          Message
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </WithStyles(TableHead)>
    <WithStyles(TableBody)>
      <WithStyles(TableRow)
        key="0"
      >
        <WithStyles(TableCell)>
          #1
          : 
          Goblin
           (
          A
          )
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          This is a test message
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
      <WithStyles(TableRow)
        key="1"
      >
        <WithStyles(TableCell)>
          #1
          : 
          Elven Ranger
           (
          B
          )
        </WithStyles(TableCell)>
        <WithStyles(TableCell)>
          This is a second test message
        </WithStyles(TableCell)>
      </WithStyles(TableRow)>
    </WithStyles(TableBody)>
  </WithStyles(Table)>
</Fragment>
`;
